optuna/importance/_fanova/_evaluator.py

```diff
@@ -0,0 +1,121 @@
+from collections import OrderedDict
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import numpy
+
+from optuna._experimental import experimental
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import DiscreteUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.importance._base import _get_distributions
+from optuna.importance._base import _get_study_data
+from optuna.importance._base import BaseImportanceEvaluator
+from optuna.importance._fanova._fanova import _Fanova
+from optuna.study import Study
+
+
+@experimental("1.3.0")
+class FanovaImportanceEvaluator(BaseImportanceEvaluator):
+    """fANOVA importance evaluator.
+
+    Implements the fANOVA hyperparameter importance evaluation algorithm in
+    `An Efficient Approach for Assessing Hyperparameter Importance
+    <http://proceedings.mlr.press/v32/hutter14.html>`_.
+
+    Given a study, fANOVA fits a random forest regression model that predicts the objective value
+    given a parameter configuration. The more accurate this model is, the more reliable the
+    importances assessed by this class are.
+
+    .. note::
+
+        Requires the `sklearn <https://github.com/scikit-learn/scikit-learn>`_ Python package.
+
+    .. note::
+
+        Pairwise and higher order importances are not supported through this class. They can be
+        computed using :class:`~optuna.importance._fanova._fanova._Fanova` directly but is not
+        recommended as interfaces may change without prior notice.
+
```

Comment:
@hvy Thank you for the insightful explanation. I understood from your reply that if the sampled hyperparameters were biased, the random forest would have poor predictive performance, so the sampling method should be changed. I agree with you. Certainly, I think it would be user-friendly to have explained in the documentation. What about adding more explanation as the note like below?

```suggestion
    .. note::
        
        The performance of fANOVA depends on the prediction performance of the underlying 
        random forest model. In order to obtain high prediction performance, it is necessary to 
        cover a wide range of the hyperparameter search space. It is recommended to use an 
        exploration-oriented sampler such as :class:`~optuna.samplers.RandomSampler`.
```

Reply:
Thanks, looks good. I applied your suggestion.